src/seccomp_fd_notify.c

```diff
@@ -130,6 +294,11 @@ void maybe_setup_seccomp_notifer() {
 
 		int notify_fd = -1;
 		notify_fd = install_notify_filter();
+
+		pthread_create(&thread_id, NULL, &catch_send_fd, (void *)&notify_fd);
+		pthread_detach(thread_id);
+	
+
 		if (send_fd(sock_fd, notify_fd) == -1) {
```

Comment:
We might miss the notify fd if catch_send_fd() isn't ready to receive notifications yet i.e. isn't in the epoll_wait loop.

Reply:
I've added an explicit synchronization variable now.. The Tini logs read like this now
`(nfsuper) ~/keerti/tsa $ docker logs e9185243013d
Connecting to the tsa.sock
Networking system calls will be intercepted
Listening to networking syscall notifications on 6..
Can send the notify fd now!
  fd=6; events: EPOLLIN
Intercepted sendmsg - passthrough!
Sent the notify fd to the seccomp agent socket at /titus-executor-sockets/titus-seccomp-agent.sock
Stopped polling notifyfd = 6!`